### PR TITLE
Add upload to NoArtifactContentUploadSerializer

### LINKED ISSUE #4348: Allow uploads to be used with NoArtifactContentUploadSerializer
Currently we allow artifact, file or upload to create single artifact content, but only file for no-artifact content.
This should allow to specify an upload to be used by both content serializers on creation.

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #4348

### CODE REVIEW

**Review by gerrod3 — COMMENTED** [review]
Have you tested this for each plugin? It looks all correct, but I worry about how interdependent this code is between core and plugins.

File: CHANGES.rst [hunk 1]
```diff
@@ -568,8 +568,7 @@ Plugin API
 Features
 ~~~~~~~~
 
-- Added ``retrieve`` logic to ``MultipleArtifactContentSerializer``. Allow to use uploads with
-  ``NoArtifactContentSerializer``.
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply CHANGES.rst:572]: It was a good one until i split the work in the original PR and forgot to adjust the changelog.

**Comment by mdellweg** [comment]
> Have you tested this for each plugin? It looks all correct, but I worry about how interdependent this code is between core and plugins.

It seems to work with pulp_rpm content related tests (`pytest -k content`).

**Comment by mdellweg** [comment]
Ansible and gem seem to fine too.

**Review by gerrod3 — APPROVED** [review]

**Review by dkliban — APPROVED** [review]

**Comment by mdellweg** [comment]
This is breaking rpm and ansible after all.
Ansible was implementing their own notion of "deferred" on the signature serializer.
RPM i have yet to check.

**Pull request merged**